docs/user/reference/environment-variables.rst

```diff
@@ -56,6 +56,31 @@ All :doc:`build processes </builds>` have the following environment variables au
     :Example: ``https://docs.readthedocs.io/ja/stable/``
     :Example: ``https://example--17.org.readthedocs.build/fr/17/``
 
+.. envvar:: READTHEDOCS_REPOSITORY_URL
+
+    URL from where the source code to generate the documentation is clonned.
+    It could be HTTPS, SSH or any other URL scheme supported by Git.
+    This is the same URL defined under :guilabel:`Admin` > :guilabel:`Settings` > :guilabel:`Repository URL`.
+
+    :Example: ``https://github.com/readthedocs/readthedocs.org``
+    :Example: ``git@github.com:readthedocs/readthedocs.org.git``
+
+.. envvar:: READTHEDOCS_REPOSITORY_IDENTIFIER
+
+    Git identifier checked out from the repository URL.
```

Comment:
Yes. It does not guarantees a commit hash, but that's on purpose. The ones that guarantees it, is the following variable.

We checkout the branch or tag name, not a commit hash unless we know the commit hash. We don't always know the commit hash before triggering the build.

Reply:
> We checkout the branch or tag name, not a commit hash unless we know the commit hash. We don't always know the commit hash before triggering the build.

This doesn't sound so failsafe when running from webhooks.

But I understand the usecase now: If there **isn't** a commit id, then it's because a build for a specific version (branch or tag) was triggered in the UI and without being attached to a specific commit :+1:

Reply:
Exactly 👍🏼